runtime/bus.go

```diff
@@ -0,0 +1,107 @@
+package runtime
+
+import (
+	"log"
+
+	"github.com/pokt-network/pocket/shared/messaging"
+	"github.com/pokt-network/pocket/shared/modules"
+	"github.com/pokt-network/pocket/telemetry"
+)
+
+var _ modules.Bus = &bus{}
+
+type bus struct {
+	modules.Bus
+
+	// Bus events
+	channel modules.EventsChannel
+
+	// Modules
+	modulesMap map[string]modules.Module
+
+	runtimeMgr modules.RuntimeMgr
+}
+
+const (
+	DefaultPocketBusBufferSize = 100
+)
+
+func CreateBus(runtimeMgr modules.RuntimeMgr) (modules.Bus, error) {
+	return new(bus).Create(runtimeMgr)
+}
+
+func (b *bus) Create(runtimeMgr modules.RuntimeMgr) (modules.Bus, error) {
+	bus := &bus{
+		channel: make(modules.EventsChannel, DefaultPocketBusBufferSize),
+
+		runtimeMgr: runtimeMgr,
+		modulesMap: make(map[string]modules.Module),
+	}
+
+	return bus, nil
+}
+
+func (m *bus) RegisterModule(module modules.Module) error {
+	m.modulesMap[module.GetModuleName()] = module
+	m.modulesMap[module.GetModuleName()].SetBus(m)
+	return nil
+}
+
+func (m *bus) PublishEventToBus(e *messaging.PocketEnvelope) {
+	m.channel <- *e
+}
+
+func (m *bus) GetBusEvent() *messaging.PocketEnvelope {
+	e := <-m.channel
+	return &e
+}
+
+func (m *bus) GetEventBus() modules.EventsChannel {
+	return m.channel
+}
+
+func (m *bus) GetPersistenceModule() modules.PersistenceModule {
+	return m.modulesMap[modules.PersistenceModuleName].(modules.PersistenceModule)
+}
+
+func (m *bus) GetP2PModule() modules.P2PModule {
+	return m.modulesMap[modules.P2PModuleName].(modules.P2PModule)
+}
+
+func (m *bus) GetUtilityModule() modules.UtilityModule {
+	return m.modulesMap[modules.UtilityModuleName].(modules.UtilityModule)
+}
+
+func (m *bus) GetConsensusModule() modules.ConsensusModule {
+	return m.modulesMap[modules.ConsensusModuleName].(modules.ConsensusModule)
+}
+
+func (m *bus) GetTelemetryModule() modules.TelemetryModule {
+	telemetryModules := []string{modules.TelemetryModuleName, "telemetry_prometheus", "telemetry_noOP"}
```

Comment:
The postfix to `_prometheus` seems like something that should come from TelemetryConfigs, but it feels like scope creep in this PR. Can you add that as a TODO and just move `_prometheus` to a constant we can reuse in a couple of places?

Reply:
Maybe addressed in https://github.com/pokt-network/pocket/pull/425/commits/488eae716d685139488fc33569dac5748935ded4 

I could have done something fancy like use an init function per implementation that appended its name to the slice in the `telemetry` package but I think it's better to KISS. When/if we'll have 10 different implementations of the same module that's gonna be another story.